lib/collection/src/lib.rs

```diff
@@ -205,6 +206,43 @@ impl Collection {
     }
 
     pub async fn update(
+        &self,
+        operation: CollectionUpdateOperations,
+        shard_selection: Option<ShardId>,
+        wait: bool,
+    ) -> CollectionResult<UpdateResult> {
+        match shard_selection {
+            Some(shard_selection) => {
+                self.update_from_peer(operation, shard_selection, wait)
+                    .await
+            }
+            None => self.update_from_client(operation, wait).await,
+        }
+    }
```

Comment:
Do we need a public general purpose `update` if we have public `update_from_client` and `update_from_peer`?

Reply:
I think you are making a good point, I will see how it looks like without the branching on the `Option` :+1:

Reply:
I have removed the general purpose update API at the collection level, now the branching takes place one level higher in the `TableOfContent`.

https://github.com/qdrant/qdrant/pull/427/commits/68633543ad7a8354e1464c703e7675c1cec8388f

Is it what you had in mind? I am not sure it is possible to push the branching higher up without code duplication.

Reply:
I actually thought to push it all the way to tonic API as there we can see that something comes from peer and so we call the appropriate method. But if it leads to a lot of code duplication then I don't mind have it as it is.